File: nodejs-wdio/tests/specs/signOut.spec.ts
import SignInPage from "../pages/signIn.page";
import { expect } from "chai";
const fs = require("fs");

describe("Sign Out Success", function () {
  before(function () {
    const file = fs.readFileSync("../fixtures/users.fixture.json");
    this.users = JSON.parse(file);
  });
  beforeEach(function () {
    // TO DO: refactor with a http post request
    SignInPage.visit();
    this.webinarsPage = SignInPage.signIn(this.users["validUser"]);
  });
  it("should destroy session", function () {
    this.webinarsPage.signOut();
    expect(SignInPage.isSignedIn(false), "Session should be destroyed").to.be
      .false;
  });
  it("should redirect to sign in page", function () {
    this.webinarsPage.signOut();
    expect(
      () => SignInPage.waitForDisplayed(),
      "Sign in page should be displayed"
    ).not.to.throw();
  });
});

File: ruby-capybara/tests/specs/sign_out_spec.rb
require 'json'
require_relative '../pages/sign_in_page'
require_relative '../pages/webinars_page'
require_relative './spec_helper.rb'

describe 'Sign Out Success', :shallow do
  before(:all) do
    file = File.read '../fixtures/users.fixture.json'
    @users = JSON.parse file
  end

  before(:each) do
    @sign_in = SignInPage.new
    @sign_in.visit
    @sign_in.sign_in(@users['validUser'])
    @webinars = WebinarsPage.new
  end

  it 'should destroy session' do
    @webinars.sign_out
    @sign_in.wait_for_displayed
    expect(@sign_in.signed_in?(false))
      .to be_falsey, 'Session should be destroyed'
  end

  it 'should redirect to sign in page' do
    @webinars.sign_out
    expect { @sign_in.wait_for_displayed }
      .not_to raise_error, 'sign in page should be displayed'
  end
end
